master/internal/kubernetes/pod.go

```diff
@@ -101,9 +153,191 @@ func (p *pod) startPod(ctx *actor.Context) error {
 	}
 }
 
-func (p *pod) configureResourcesRequirements() v1.ResourceRequirements {
-	return v1.ResourceRequirements{
-		Limits: map[v1.ResourceName]resource.Quantity{
+func (p *pod) receivePodStatusUpdate(ctx *actor.Context, msg podStatusUpdate) error {
+	p.pod = msg.updatedPod
+
+	switch msg.updatedPod.Status.Phase {
+	case k8sV1.PodPending:
+		// When pods are deleted, Kubernetes sometimes transitions pod statuses
+		// to pending prior to deleting them. We ignore this state if we have
+		// triggered the deletion.
+		if p.resourcesDeleted {
+			ctx.Log().WithField("pod", p.podName).Info(
+				"ignoring pod status because resources are being deleted")
+			return nil
+		}
+
+		containerState := getContainerState(msg.updatedPod.Status.Conditions)
+		if containerState == container.Running {
+			ctx.Log().WithField("pod", p.podName).Errorf(
+				"unexpected containers status while pod is pending")
+		}
+
+		if containerState != p.container.State {
```

Comment:
non-blocking: For indentation reduction's sake, perhaps something like
```go
if containerState == p.container.State {
  return nil
}
```
here and in other places below.

Reply:
done